server/models/Collective.js

```diff
@@ -1261,6 +1262,19 @@ Collective.prototype.deactivateAsHost = async function () {
     );
   }
 
+  // Make sure we clean up all pending applications
+  await models.HostApplication.update(
+    { status: HostApplicationStatus.EXPIRED },
+    { where: { HostCollectiveId: this.id } },
+  );
+
+  await models.Member.update({ deletedAt: new Date() }, { where: { MemberCollectiveId: this.id, role: 'HOST' } });
```

Comment:
:ice_cream: Not a requested change, just a cleaner option IMO

```suggestion
  await models.Member.destroy({ where: { MemberCollectiveId: this.id, role: 'HOST' } });
```